xdsl/interfaces.py

```diff
@@ -49,3 +50,35 @@ class HasCanonicalizationPatternsInterface(Operation, abc.ABC):
     @abc.abstractmethod
     def get_canonicalization_patterns(cls) -> tuple[RewritePattern, ...]:
         raise NotImplementedError()
+
+
+class _HasFolderInterfaceTrait(OpTrait):
+    # TODO: Do we even need a trait if there's already an interface?
+    def verify(self, op: Operation) -> None:
+        return
```

Comment:
`HasCanonicalizationInterface` does something similar but that's maybe just for backwards compatibility?

Reply:
Yeah all traits need a verify method but sometimes it's a noop